include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -2684,251 +3266,505 @@ FEPointEvaluation<n_components_, dim, spacedim, Number>::do_integrate(
   const StridedArrayView<ScalarNumber, stride_view> &solution_values,
   const EvaluationFlags::EvaluationFlags            &integration_flags,
   const bool                                         sum_into_values)
 {
-  do_integrate<true>(solution_values, integration_flags, sum_into_values);
+  if (!sum_into_values && this->fe->n_components() > n_components)
+    for (unsigned int i = 0; i < solution_values.size(); ++i)
+      solution_values[i] = 0;
+
+  do_integrate_in_face<do_JxW, is_linear, 1>(this->scratch_data_scalar.begin(),
+                                             integration_flags,
+                                             false);
+
+  ScalarNumber *input = this->scratch_data_scalar.begin();
+
+  if (stride_view == 1 && this->component_in_base_element == 0 &&
+      (is_linear || this->renumber.empty()))
+    {
+      if (sum_into_values)
+        internal::
+          FEFaceNormalEvaluationImpl<dim, is_linear ? 1 : -1, ScalarNumber>::
+            template interpolate<false, true>(n_components,
+                                              integration_flags,
+                                              this->shape_info,
+                                              input,
+                                              solution_values.data(),
+                                              this->current_face_number);
+      else
+        internal::
+          FEFaceNormalEvaluationImpl<dim, is_linear ? 1 : -1, ScalarNumber>::
+            template interpolate<false, false>(n_components,
+                                               integration_flags,
+                                               this->shape_info,
+                                               input,
+                                               solution_values.data(),
+                                               this->current_face_number);
+    }
+  else
+    {
+      const unsigned int dofs_per_comp_face =
+        is_linear ? Utilities::pow(2, dim - 1) : this->dofs_per_component_face;
+
+      const unsigned int size_input = 3 * dofs_per_comp_face * n_components;
+      ScalarNumber      *output     = input + size_input;
+
+      internal::
+        FEFaceNormalEvaluationImpl<dim, is_linear ? 1 : -1, ScalarNumber>::
+          template interpolate<false, false>(n_components,
+                                             integration_flags,
+                                             this->shape_info,
+                                             input,
+                                             output,
+                                             this->current_face_number);
+
+      const unsigned int dofs_per_comp =
+        is_linear ? Utilities::pow(2, dim) : this->dofs_per_component;
+
+      for (unsigned int comp = 0; comp < n_components; ++comp)
+        {
+          const std::size_t offset =
+            (this->component_in_base_element + comp) * dofs_per_comp;
```

Comment:
It should be considered here @jh66637 and tests are passing for me. Do you have a minimal example?

Reply:
I added a failing test here: https://github.com/dealii/dealii/compare/master...jh66637:dealii:fcl

Reply:
Thanks for the test. It did't compile but I got what was wrong 😉 The strides are fixed and it should work now. This was some nasty debugging. Can you check @jh66637 ?

Reply:
Sorry, this didn't compile out of the box, and thank you for the fix. I tested this in the Nitsche-type mortaring tutorial and can confirm it is now working as expected :+1: